parsec/core/gui/navigation_bar_widget.py

```diff
@@ -0,0 +1,128 @@
+        self.scroll_area.setFrameShadow(QFrame.Shadow.Plain)
+        self.scroll_area.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.scroll_area.setLineWidth(0)
+        self.scroll_area.setFixedHeight(25)
+        self.scroll_area.setWidgetResizable(True)
+        self.scroll_area.horizontalScrollBar().rangeChanged.connect(self._on_range_changed)
+        self.layout().addWidget(self.scroll_area)
+        self.inner_widget = QWidget()
+        self.inner_widget.setFixedHeight(25)
+        self.inner_widget.setLayout(QHBoxLayout())
+        self.scroll_area.setWidget(self.inner_widget)
+        self.inner_widget.layout().setContentsMargins(0, 3, 0, 0)
+        self.inner_widget.layout().addItem(QSpacerItem(10, 0, QSizePolicy.Policy.MinimumExpanding))
+        self.paths = []
+        self.workspace_name = None
+
+    def _on_range_changed(self, min, max):
+        self.scroll_area.horizontalScrollBar().setSliderPosition(max)
+
+    def _on_label_clicked(self, idx):
+        def _internal_label_clicked(_title):
+            complete_path = FsPath([EntryName(p) for i, p in enumerate(self.paths) if i < idx])
+            self.route_clicked.emit(complete_path)
+
+        return _internal_label_clicked
+
+    def get_current_path(self):
+        return FsPath(self.paths)
+
+    def from_path(self, workspace_name, path):
+        self.clear()
+        path = FsPath(path)
+        parts = [workspace_name]
+        self.workspace_name = workspace_name
+        parts.extend(path.parts)
+        self.paths = parts[1:]
+        for idx, part in enumerate(parts):
+            icon = IconLabel()
+            icon.setScaledContents(True)
+            icon.setProperty("color", QColor(153, 153, 153))
+            if idx == 0:
+                # Adding a new spacer item creates problems so we use the
+                # icon to add a margin
+                icon.setFixedSize(15, 10)
+                icon.setPixmap(QPixmap(":/icons/images/material/fiber_manual_record.svg"))
+                icon.apply_style()
+                icon.setStyleSheet("margin-right: 5px;")
+                self.layout().insertWidget(0, icon)
+            else:
+                icon.setFixedSize(15, 15)
+                icon.setPixmap(QPixmap(":/icons/images/material/chevron_right.svg"))
+                icon.apply_style()
+                self.inner_widget.layout().insertWidget(
+                    self.inner_widget.layout().count() - 1, icon
+                )
+            label = None
+            if idx != len(parts) - 1:
+                obj_name = part.replace(" ", "_")
+                label = ClickableLabel(part)
+                label.clicked.connect(self._on_label_clicked(idx))
+                label.setObjectName(f"label_{obj_name}")
+                label.setStyleSheet(
+                    f"#label_{obj_name} {{ color: #999999; }} #label_{obj_name}:hover {{ color: #0092FF; }}"
+                )
+            else:
```

Comment:
Is this `if-else` really necessary? I mean clicking the current directory seems fine to me

Reply:
It's just that clicking it does absolutely nothing so I would rather avoid showing it as clickable.